spark/src/main/scala/org/apache/comet/serde/QueryPlanSerde.scala

```diff
@@ -2428,6 +2428,22 @@ object QueryPlanSerde extends Logging with ShimQueryPlanSerde with CometExprShim
           withInfo(expr, "unsupported arguments for ArrayJoin", exprs: _*)
           None
         }
+      case expr @ ArrayFilter(child, _) if ArrayCompact(child).replacement.sql == expr.sql =>
```

Comment:
Would you add a flag to enable and disable?

Reply:
Because the PR only contains basic tests, could you add a check to enable this expression only if `CometConf.COMET_CAST_ALLOW_INCOMPATIBLE` is enabled? We can remove this check in a future PR that adds comprehensive tests and demonstrates that we have Spark-compatible behavior for all supported data types.

Reply:
COMET_CAST_ALLOW_INCOMPATIBLE check has been added.